using-secrets_5b8506120428631d7a8a7239.md

```diff
@@ -387,17 +413,69 @@ file of that secret and execute `sem create`.
 
 In that case the pipeline will fail to execute.
 
+## An example
+
+The following `secret` defines two environment variables and one file:
+
+    apiVersion: v1beta
+    kind: Secret
+    metadata:
+      name: ssh-keys
+    data:
+      env_vars:
+        - name: SSH_KEY_1
+          value: SSH Secret key
+        - name: SSH_KEY_2
+          value: Secret SSH key TWO
+    files:
+    - path: file.txt
+      content: SGVsbG8gU2VtYXBob3JlIDIuMCEK
+
 ## Use cases
 
-This section will present some real world scenarios that use Semaphore 2.0 and
-utilize secrets.
+This section will present a real world scenarios that use Semaphore 2.0 and
+utilize secrets to connect to a server using SSH.
 
 ### How to handle a ssh key (connect to remote server)
 
 In this section you will learn how to store and retrieve ssh keys as secrets
 in order to connect to a remote SSH server without needing a password or
 exchange files securely using SFTP without the need to login.
 
+The `secrets` YAML file will be as follows:
+
+    apiVersion: v1beta
+    kind: Secret
+    metadata:
+      name: ssh-keys
+    data:
+      env_vars: []
+      files:
+      - path: .ssh/server_key
+        content: SGVsbG8gU2VtYXBob3JlIDIuMCEK
+
+The pipeline YAML file will be as follows:
+
+	version: v1.0
+	name: Using env_vars per jobs
+	agent:
+	  machine:
+	    type: e1-standard-2
+	    os_image: ubuntu1804
+    
+	blocks:
+	  - name: Using SSH key files
+	    task:
+	      jobs:
+	      - name: Using server_key
+	        commands:
+	          - ls -l ~/.ssh
+			  - chmod 600 ~/.ssh/server_key
```

Comment:
By looking at this, I might even argue that you should be able to define the file mode in the secret. cc @mactsouk @darkofabijan let me know what you think about that.

Reply:
It is not a bad idea - or maybe keep the file permissions of the original file.
On the other hand, using `chmod` is not a big issue.